www/__tests__/metricsHelper.test.ts

```diff
@@ -70,59 +75,264 @@ describe('metricsHelper', () => {
+            ['value1', 10],
+            ['value2', 20],
+          ],
+        },
+        {
+          key: 'label_mode2',
+          values: [
+            ['value3', 30],
+            ['value4', 40],
+          ],
+        },
+      ];
+      const result = isCustomLabels(modeMap);
+      expect(result).toBe(true);
+    });
+
+    it('returns false for mixed custom and sensed labels', () => {
+      const modeMap = [
+        {
+          key: 'label_mode1',
+          values: [
+            ['value1', 10],
+            ['value2', 20],
+          ],
+        },
+        {
+          key: 'MODE2',
+          values: [
+            ['value3', 30],
+            ['value4', 40],
+          ],
+        },
+      ];
+      const result = isCustomLabels(modeMap);
+      expect(result).toBe(false);
+    });
+  });
+
+  describe('isAllCustom', () => {
+    it('returns true when all keys are custom', () => {
+      const isSensedKeys = [false, false, false];
+      const isCustomKeys = [true, true, true];
+      const result = isAllCustom(isSensedKeys, isCustomKeys);
+      expect(result).toBe(true);
+    });
+
+    it('returns false when all keys are sensed', () => {
+      const isSensedKeys = [true, true, true];
+      const isCustomKeys = [false, false, false];
+      const result = isAllCustom(isSensedKeys, isCustomKeys);
+      expect(result).toBe(false);
+    });
+
+    it('returns undefined for mixed custom and sensed keys', () => {
+      const isSensedKeys = [true, false, true];
+      const isCustomKeys = [false, true, false];
+      const result = isAllCustom(isSensedKeys, isCustomKeys);
+      expect(result).toBe(undefined);
+    });
+  });
+
+  describe('getUnitUtilsForMetric', () => {
+    const imperialConfig = {
+      distanceSuffix: 'mi',
+      speedSuffix: 'mph',
+      convertDistance: jest.fn((d) => d),
+      convertSpeed: jest.fn((s) => s),
+      getFormattedDistance: jest.fn((d) => `${d} mi`),
+      getFormattedSpeed: jest.fn((s) => `${s} mph`),
+    };
+
+    it('checks for distance metric', () => {
+      const result = getUnitUtilsForMetric('distance', imperialConfig);
+      expect(result).toEqual(['mi', expect.any(Function), expect.any(Function)]);
+      expect(result[1](1)).toBe(1);
+      expect(result[2](1)).toBe('1 mi mi');
+    });
+
+    it('checks for duration metric', () => {
+      const result = getUnitUtilsForMetric('duration', imperialConfig);
+      expect(result).toEqual(['hours', expect.any(Function), expect.any(Function)]);
+      expect(result[1](3600)).toBe(1);
+      expect(result[2](3600)).toBe('1 hours');
+    });
+
+    it('checks for count metric', () => {
+      const result = getUnitUtilsForMetric('count', imperialConfig);
+      expect(result).toEqual(['trips', expect.any(Function), expect.any(Function)]);
+      const mockTrip = { responded: 4, not_responded: 3 };
+      expect(result[1](mockTrip)).toBe(mockTrip);
+      expect(result[2](mockTrip)).toBe(mockTrip + ' trips');
+    });
```

Comment:
This passes, but I don't think it is testing the right functionality. `count` is a metric to just count the number of trips. The metric that expects a structure with `responded` and `not_responded` is **`response_count`**